libexec/goenv-version-file-read

```diff
@@ -10,12 +10,16 @@ if [ ! -f "$VERSION_FILE" ]; then
   exit 1
 fi
 
-# NOTE: Read the first non-whitespace word from the specified version file.
-# Be careful not to load it whole in case there's something crazy in it.
-IFS="${IFS}"$'\r'
-words=($(cut -b 1-1024 "$VERSION_FILE" | sed -n 's/^[[:space:]]*\([^[:space:]#][^[:space:]]*\).*/\1/p'))
+if [[ "$(basename $VERSION_FILE)" == "go.mod" ]]; then
+  versions=("$(cat "$VERSION_FILE" | sed -En 's/^go[[:space:]]+([[:digit:].]+)$/\1/p')")
```

Comment:
Why not this? Filter out all the noise from the go.mod file first:

`versions=($(cat go.mod | grep -E "go \\d\\.\\d{1,2}(beta|rc)?" | sed 's/go //'`))`

Reply:
See below - thank you :)